src/components/StatusIndicator/StatusIndicator.tsx

```diff
@@ -0,0 +1,82 @@
+import * as PropTypes from 'prop-types'
+import * as React from 'react'
+import { Icon } from '../../'
+
+import { customPropTypes, UIComponent, createShorthandFactory } from '../../lib'
+import { ComponentVariablesInput, IComponentPartStylesInput } from '../../../types/theme'
+import { Extendable, ItemShorthand } from '../../../types/utils'
+
+export interface IStatusIndicatorProps {
+  as?: any
+  className?: string
+  size?: number
+  icon?: ItemShorthand
+  styles?: IComponentPartStylesInput
+  variables?: ComponentVariablesInput
+}
+
+/**
+ * A status indicator is a graphical representation of user's status
+ */
+class StatusIndicator extends UIComponent<Extendable<IStatusIndicatorProps>, any> {
+  static create: Function
+
+  static className = 'ui-statusindicator'
```

Comment:
We may have a shortcoming in our conformance tests that would fail this, but the `className` should be `status-indicator` since it is multiple words.

Reply:
yes, status-indicator results in test failure. Will keep it this way for now and we can fix it later.

---
In reply to: [215421511](https://github.com/stardust-ui/react/pull/168#discussion_r215421511) [](ancestors = 215421511)